cbmc/patches/0001-Modify-MQTT-to-let-CBMC-proofs-stub-out-RemoveAllMat.patch

```diff
@@ -1,17 +1,46 @@
-From a1455991c02aa6d7db1d4333e2cdc5add392c105 Mon Sep 17 00:00:00 2001
-From: Mark R Tuttle <mrtuttle@amazon.com>
-Date: Thu, 13 Feb 2020 17:39:51 +0000
-Subject: [PATCH] Modify MQTT to let CBMC proofs stub out RemoveAllMatches
-
----
- libraries/standard/common/include/iot_linear_containers.h | 8 ++++++++
- 1 file changed, 8 insertions(+)
-
 diff --git a/libraries/standard/common/include/iot_linear_containers.h b/libraries/standard/common/include/iot_linear_containers.h
-index 689ea5e..ffb3469 100644
+index 689ea5e..09061f2 100644
 --- a/libraries/standard/common/include/iot_linear_containers.h
 +++ b/libraries/standard/common/include/iot_linear_containers.h
-@@ -719,6 +719,13 @@ static inline IotLink_t * IotListDouble_RemoveFirstMatch( const IotListDouble_t
+@@ -489,6 +489,9 @@ static inline void IotListDouble_InsertSorted( IotListDouble_t * const pList,
+  * @param[in] pLink The element to remove.
+  */
+ /* @[declare_linear_containers_list_double_remove] */
++#ifdef CBMC_STUB_REMOVE
```

Comment:
Would it be better to call it ```CBMC_STUB_IOT_LIST_DOUBLE_REMOVE``` and  ```CBMC_STUB_IOT_LIST_DOUBLE_REMOVE_ALL``` to avoid possible clash.

Reply:
I'm going to leave this to the clean up phase after all proofs are in place and we can look at them all together.  (This change will require changes to several other proofs.)